src/core/lombok/javac/apt/LombokFileObjects.java

```diff
@@ -152,130 +142,28 @@ static Compiler getCompiler(JavaFileManager jfm) {
-			URI uri = fileObject.toUri();
-			if (uri.getScheme() == null) {
-				uri = URI.create("file:///" + uri);
-			}
-			try {
-				return Paths.get(uri);
-			} catch (IllegalArgumentException e) {
-				throw new IllegalArgumentException("Problems in URI '" + uri + "' (" + fileObject.toUri() + ")", e);
-			}
-		}
-		
-		private static BaseFileManager asBaseFileManager(JavaFileManager jfm) {
-			if (jfm instanceof BaseFileManager) {
-				return (BaseFileManager) jfm;
-			}
-			return new FileManagerWrapper(jfm);
-		}
-		
-		static class FileManagerWrapper extends BaseFileManager {
-			JavaFileManager manager;
-			
-			public FileManagerWrapper(JavaFileManager manager) {
-				super(null);
-				this.manager = manager;
-			}
-			
-			@Override
-			public int isSupportedOption(String option) {
-				return manager.isSupportedOption(option);
-			}
-			
-			@Override
-			public ClassLoader getClassLoader(Location location) {
-				return manager.getClassLoader(location);
-			}
-			
-			@Override
-			public Iterable<JavaFileObject> list(Location location, String packageName, Set<Kind> kinds, boolean recurse) throws IOException {
-				return manager.list(location, packageName, kinds, recurse);
-			}
-			
-			@Override
-			public String inferBinaryName(Location location, JavaFileObject file) {
-				return manager.inferBinaryName(location, file);
-			}
-			
-			@Override
-			public boolean isSameFile(FileObject a, FileObject b) {
-				return manager.isSameFile(a, b);
-			}
-			
-			@Override
-			public boolean handleOption(String current, Iterator<String> remaining) {
-				return manager.handleOption(current, remaining);
-			}
-			
-			@Override
-			public boolean hasLocation(Location location) {
-				return manager.hasLocation(location);
-			}
-			
-			@Override
-			public JavaFileObject getJavaFileForInput(Location location, String className, Kind kind) throws IOException {
-				return manager.getJavaFileForInput(location, className, kind);
-			}
-			
-			@Override
-			public JavaFileObject getJavaFileForOutput(Location location, String className, Kind kind, FileObject sibling) throws IOException {
-				return manager.getJavaFileForOutput(location, className, kind, sibling);
-			}
-			
-			@Override
-			public FileObject getFileForInput(Location location, String packageName, String relativeName) throws IOException {
-				return manager.getFileForInput(location, packageName, relativeName);
-			}
-			
-			@Override
-			public FileObject getFileForOutput(Location location, String packageName, String relativeName, FileObject sibling) throws IOException {
-				return manager.getFileForOutput(location, packageName, relativeName, sibling);
-			}
-			
-			@Override
-			public void flush() throws IOException {
-				manager.flush();
-			}
-			
-			@Override
-			public void close() throws IOException {
-				manager.close();
-			}
+		} catch (NoSuchMethodException e) {
+			return null;
+		} catch (InvocationTargetException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)